### ISSUE TITLE
Crazy amount of dependencies pulled in by `image` crate.

### ISSUE THREAD

**Issue by virtualritz** [atanunq/viuer#61]
I had this fixed in #59 but you reverted it, I think?

Changing 
```toml
[dependencies]
image = "0.25"
```

to

```toml
[dependencies]
image = { version = "0.25", default-features = false, features = [
    "rayon",
    "png",
] }
```

brings down the number of deps from 129 to 61 (when `--all-features` is passed)!

But even w/o `--all-features` it is 54 vs 118. None of these dependencies are needed for `viuer`
This add *considerable* compile time for zero gain when depending on `viuer.`

Pretty please, roll this back.

**Comment by virtualritz** [comment]
This should not be "basic `viuer` functionality". Basic `viuer` functionality should be to take an `Image::DynamicImage` and display that. Nothing more.

A user can add the `image` to their `Cargo.toml` to add support for formats they need. I.e. TBH even the `dep:image:png` feature should actually only be in `dev-dependencies`.

Also consider someone may _not_ want to use `image` at all.

E.g. my `openimageio` Rust wrapper also supports  `Into<image::DynamicImage>`.

So a user could use that to read the a JPG or OpenEXR (much better support than e.g. the `image` crate), do a `Into<image::DynamicImage>` and then use `viuer` to print that.

TLDR; you are assumung which image formats a user fo your crate wants to load. Your crate should be about displaying images, not loading image data.

**Comment by virtualritz** [comment]
That said, you can add a bunch of features and forward them to the `image` crate in `viuer`'s `Cargo.toml`.

While this saves users from pulling in `image` themselves, it doesn't really do anything. Basically, a user would have:

I also think the `image` crate pulling in all these images by default is not great. Many crates depend on `image` but many only use a fraction of the crate or need a specific format. The default config adds about 50 dependencies to a build.

Build times on Rust are an issue. For `openimageio` when build the `test`s which depend on `viuer` which depends on `image` with default features, this almost doubles the build times, needlessly.

**Comment by virtualritz** [comment]
So if you wanted to do this, it would do it like like this:

```toml
[package]
name = "viuer"
version = "0.8.1"
authors = ["Atanas Yankov <atanas.yankov98@gmail.com>"]
edition = "2021"
description = "Display images in the terminal"
documentation = "https://docs.rs/viuer"
readme = "README.md"
repository = "https://github.com/atanunq/viuer"
license = "MIT"
categories = ["command-line-interface"]
keywords = ["terminal", "image"]
exclude = [".github"]

[features]
default-features = ["default-formats", "rayon"]
default-formats = ["bmp", "gif", "jpeg", "tga", "tiff"]
avif = ["image/avif"]
bmp = ["image/bmp"]
dds = ["image/dds"]
exr = ["image/exr"]
ff = ["image/ff"]
gif = ["image/gif"]
hdr = ["image/hdr"]
ico = ["image/ico"]
jpeg = ["image/jpeg"]
png = ["image/png"]
pnm = ["image/pnm"]
qoi = ["image/qoi"]
tga = ["image/tga"]
tiff = ["image/tiff"]
webp = ["image/webp"]
rayon = ["image/rayon"]
sixel = ["sixel-rs"]

[dependencies]
ansi_colours = "1"
base64 = "0.22"
console = { version = "0.15", default-features = false }
crossterm = { version = "0.28", default-features = false }
image = { version = "0.25", default-features = false, features = ["png"] }
lazy_static = "1.5"
sixel-rs = { version = "0.3", optional = true }
tempfile = "3"
termcolor = "1"

[target.'cfg(windows)'.dependencies]
crossterm = { version = "0.28", default-features = false, features = [
    "windows",
] }
```

But now you need to keep image format features in sync between `image` and `viuer`. Whereas if you let the user add the `image` dependency themselves (i.e. as I suggested above), you don't have this additional maintenance burden.

**Comment by atanunq** [comment]
My preference is that `viuer` continues to support its `print_from_file` functionality that does the image loading behind the scenes. I do not want to force all users to use the `image` crate. Nevertheless, I agree that there should exist a flag to reduce unneeded dependencies. Thank you very much for providing the example `Cargo.toml`. I will give it a shot and release a new version in the next few days.

**Comment by virtualritz** [comment]
Another option is to put `print_from_file` behind a feature flag that enables a `default-formats` and is off by default (`default-formats` would also be off by default).

That would mean a crate user can get the comfort buy simply enabling that feature flag but a normal user of the crate isn't adding 50-
–60 dependencies to their build.

**Comment by atanunq** [comment]
Really liked that idea! Please check out 0.9 and let me know if that works for you

**Comment by virtualritz** [comment]
Two things: is the `png` feature still needed on `image`?
I think `image` should have `rayon` on by default, I think.
